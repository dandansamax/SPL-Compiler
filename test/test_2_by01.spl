struct Pokemon{
    char name[20];
    int hp;
};

Pokemon get_pokemon(int id){
    struct Pokemon pikachu;
    pikachu.name="Pikachu";
    pikachu.hp=50;
    return pikachu;
}

Pokemon set_pokemon(int id){
    struct Pokemon pikachu;
    pikachu.name="Pikachu";
    pikachu.hp=50;
    return pikachu;
}

int test_2_by01(){
    Pokemon who=get_pokemon(1);
    get_pokemon=set_pokemon;
}