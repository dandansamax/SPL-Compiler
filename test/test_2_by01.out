Error type 5 at Line 15: unmatching types appear at both sides of the assignment operator (=): LHS is struct Pokemon, but RHS is int.
